src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -83,6 +87,17 @@ def _pull_image(image_str, job_dir):
                              f"Standard error: {std_err.read()}")
 
 
+def _write_to_file(file_path, content, mode="x"):
+    """
+    Writes the specified content to the specified file.
+    """
+    try:
+        with open(file_path, mode) as f:
+            f.write(content)
+    except FileExistsError:
+        print(f"File {file_path} already exists.")
```

Comment:
This makes me very nervous as it could cause difficult to find bugs if files are overwritten but people aren't carefully checking the logs. I'd prefer to throw the error by default and have a flag to ignore file collisions

Reply:
👍

Reply:
I meant there should be a cli flag to overwrite existing files

Reply:
that's done by giving the 'w' mode.

Reply:
That's not a CLI flag?

Reply:
Thinking about this further, IIUC if any of the submission scripts / text files exist, they won't be overwritten by this logic. I guess I don't understand why you wouldn't just generate all new scripts (maybe backing up the old ones if necessary)

Reply:
My aim was to avoid overwriting existing files, particularly in situations where a user runs the script multiple times. For instance, if a user executes the script again without specifying the --force option, it will overwrite all existing files. This may not be problematic if all the arguments remain the same. However, if the user runs the script with a different source_dir argument and without --force, the situation becomes complicated.

Reply:
I feel like I'm missing something here - IIUC if I run the script twice in a row with different arguments, the script will ignore the 2nd set of arguments because it doesn't overwrite the files. That seems like the exact opposite of the behavior anyone would expect

Reply:
Maybe we should zoom on Monday and talk through this, this thread is getting long

Reply:
👍 made a few changes. But sure let's sync up on Monday.

Reply:
Files are always overwritten if the script proceeds at this point so moot

:+1: